### PR TITLE
dot/core, lib/transaction: remove included extrinsics from queue

### LINKED ISSUE #329: Remove Included Extrinsics from Queue
When a block is produced by someone else and received, we need to remove the included extrinsics from the extrinsics queue. Sorting by priority may make this easier, but we may also need a more complex data structure to provide quick lookups/removals.

### PR DESCRIPTION
**Pull request by noot**
## Changes
- use transaction pool hashmap in transaction queue to easily determine if extrinsic is in queue or not
- update transaction queue to have RemoveExtrinsics function
- call this function upon block response that includes block body

## Tests:
```
go test ./dot/core/ ./lib/transaction
```

### Issues:
closes #329

### CODE REVIEW

File: dot/core/sync.go [hunk 1]
```diff
@@ -284,3 +276,63 @@ func (s *Syncer) handleBlockResponse(msg *network.BlockResponseMessage) (int64,
 
 	return highestInResp, nil
 }
+
+// handleHeader handles headers included in BlockResponses
+func (s *Syncer) handleHeader(header *types.Header) (int64, error) {
+	highestInResp := int64(0)
+
+	// get block header; if exists, return
+	// TODO: update blockState to include Has function
```
Comment (ryanchristo) [inline dot/core/sync.go:285]: issue number?

File: dot/core/sync.go [hunk 2]
```diff
@@ -284,3 +276,63 @@ func (s *Syncer) handleBlockResponse(msg *network.BlockResponseMessage) (int64,
...
+	existingHeader, err := s.blockState.GetHeader(header.Hash())
+	if err != nil && existingHeader == nil {
+		err = s.blockState.SetHeader(header)
+		if err != nil {
+			return 0, err
+		}
+
+		log.Info("[sync] saved block header", "hash", header.Hash(), "number", header.Number)
+
+		// TODO: handle consensus digest, if first in epoch
```
Comment (ryanchristo) [inline dot/core/sync.go:295]: #599 / #703

File: dot/core/sync.go [hunk 3]
```diff
@@ -284,3 +276,63 @@ func (s *Syncer) handleBlockResponse(msg *network.BlockResponseMessage) (int64,
...
+	for _, ext := range exts {
+		s.transactionQueue.RemoveExtrinsic(ext)
+	}
+
+	return err
+}
+
+// handleHeader handles blocks (header+body) included in BlockResponses
+func (s *Syncer) handleBlock(block *types.Block) error {
+	// TODO: execute block and verify authorship right
```
Comment (ryanchristo) [inline dot/core/sync.go:322]: issue number?

Comment (noot) [reply dot/core/sync.go:322]: #681  #684

File: dot/core/sync.go (hunk 1 above)
Comment (noot) [reply dot/core/sync.go:285]: #750

**Pull request merged**